src/io/export_pov.cc

```diff
@@ -55,18 +56,47 @@ void export_pov(const std::shared_ptr<const Geometry>& geom, std::ostream& outpu
 
   auto has_color = !ps->color_indices.empty();
 
+  auto new_set = PolySet::createEmpty();
+
   for (size_t polygon_index=0; polygon_index<ps->indices.size(); polygon_index++) {
     const auto &polygon = ps->indices[polygon_index];
-    output << "polygon { " << polygon.size() + 1 << ", \n";
-    for (size_t i=0; i<polygon.size(); i++) {
-      if (i)
-        output << ", ";
-      const auto & x = ps->vertices[polygon[i]].x();
-      const auto & y = ps->vertices[polygon[i]].y();
-      const auto & z = ps->vertices[polygon[i]].z();
-      output << "<" << x << ", " << y << ", " << z << ">";
+    // create a PolySetBuilder with only this polygon
+    PolySetBuilder psb;
+    psb.beginPolygon(polygon.size());
+    for (size_t i=0; i<polygon.size(); i++)
+      psb.addVertex(ps->vertices[polygon[i]]);
+    auto temp_polyset = psb.build();
+    // triangulate it
+    temp_polyset = PolySetUtils::tessellate_faces(*temp_polyset);
```

Comment:
Can't you just call `PolySetUtils::tessellate_faces()` on the initial polyset, then iterate over triangles?

Reply:
Then you loose context. What I mean with that you no longer know which triangles are part of the same mesh.

Reply:
Something feels wrong.

One PolySet conceptually should map to one mesh. When you call tessellate_faces, you get the same geometry and color information back, just with the guarantee that each polygon is a triangle.

It's possible that this is a result of you creating one PovRay mesh per _polygon_ instead of per _PolySet_.
Take a look at https://www.povray.org/documentation/3.7.0/r3_4.html#r3_4_5_2_4
It describes that you can create indexed meshes where you specify a texture index for each triangle:

```
face_indices {
    2,
    <0, 1, 2>, 0,
    <3, 4, 5>, 1,

..etc.
```